include/openmc/distribution_spatial.h

```diff
@@ -136,6 +136,34 @@ class MeshSpatial : public SpatialDistribution {
                                 //!< mesh element indices
 };
 
+//==============================================================================
+//! Distribution of points
+//==============================================================================
+
+class PointCloud : public SpatialDistribution {
+public:
+  explicit PointCloud(pugi::xml_node node);
+  explicit PointCloud(gsl::span<const double> x, gsl::span<const double> y,
+    gsl::span<const double> z, gsl::span<const double> strengths);
+
+  //! Sample a position from the distribution
+  //! \param seed Pseudorandom number seed pointer
+  //! \return Sampled position
+  Position sample(uint64_t* seed) const override;
+
+  //! Sample a point
+  //! \param seed Pseudorandom number seed pointer
+  //! \return Sampled point index
+  int32_t sample_point_index(uint64_t* seed) const;
+
+  double total_strength() { return this->point_idx_dist_.integral(); }
+
+private:
+  gsl::span<const double> x_, y_, z_;
```

Comment:
could assemble these into a vector of `Position` on initialization, but not sure pros/cons of that choice vs. keeping these separate...

Reply:
One more thought, we should make sure to use a container here so the object owns the memory of these attributes -- `gsl::span` is just a view into memory space and doesn't "own" these values.